pkg/scheduler/framework/plugins/interpodaffinity/plugin.go

```diff
@@ -221,9 +221,33 @@ func (pl *InterPodAffinity) isSchedulableAfterNodeChange(logger klog.Logger, pod
 		return framework.Queue, err
 	}
 
+	// When queuing this Pod:
+	// - 1. A new node is added with the pod affinity topologyKey, the pod may become schedulable.
+	// - 2. The original node does not have the pod affinity topologyKey but the modified node does, the pod may become schedulable.
+	// - 3. Both the original and modified nodes have the pod affinity topologyKey and they differ, the pod may become schedulable.
 	for _, term := range terms {
-		if _, ok := modifiedNode.Labels[term.TopologyKey]; ok {
-			logger.V(5).Info("a node with matched pod affinity topologyKey was added/updated and it may make pod schedulable",
+		if originalNode == nil {
+			if _, ok := modifiedNode.Labels[term.TopologyKey]; ok {
+				// Case 1: A new node is added with the pod affinity key.
+				logger.V(5).Info("A node with a matched pod affinity topologyKey was added and it may make the pod schedulable",
+					"pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
+				return framework.Queue, err
+			}
+			continue
+		}
+		originalTopologyValue, originalHasKey := originalNode.Labels[term.TopologyKey]
+		modifiedTopologyValue, modifiedHasKey := modifiedNode.Labels[term.TopologyKey]
+
+		if !originalHasKey && modifiedHasKey {
+			// Case 2: Original node does not have the pod affinity key, but the modified node does.
+			logger.V(5).Info("A node with a matched pod affinity topologyKey was updated and it may make the pod schedulable",
+				"pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
+			return framework.Queue, err
+		}
+
+		if originalHasKey && modifiedHasKey && originalTopologyValue != modifiedTopologyValue {
+			// Case 3: Both nodes have the pod affinity key, but the values differ.
```

Comment:
In this case, I am not sure whether it is effective, because when we modify the TopologyKey of the node, if it is the same but the value is different, will this be effective for pod affinity? @sanposhiho @dom4ha

Reply:
what do you mean by "effective"? the current implementation here for case 3 looks correct to me

Reply:
But does the value of topologyKey mean anything for podaffinity here? In my opinion, the only case when podaffinity scheduling fails is when there is no node with the topologyKey. If the new node is added or updated to have a matching topologyKey, we should requeue the pod. Changing the value means nothing, as we could always be able to schedule the pod on the previous node then.

Reply:
> does the value of topologyKey mean anything for podaffinity here? 

It does.
Let's say only node1 has a matching pod, and it's in domain1. So, all nodes in domain1 can get this pod. But, then let's say the pod isn't schedulable due to other reasons, e.g., no enough space in domain1.
Then, now the node1 gets a label update and moved to domain2. This change makes all nodes in domain2 schedulable, and maybe the domain2 has enough space for the pod.

Reply:
Right, thanks for explanation.